src/runtime/statistics.py

```diff
@@ -220,6 +220,64 @@ def harmonic_mean(x: list[f64]) -> f64:
 
     return k / sum
 
+
+def mode(x: list[i32]) -> i32:
+    """
+    Returns the mode of a data sequence of numbers
+    """
+    k: i32 = len(x)
+    if k == 0:
+        raise Exception("No mode for empty data")
+    ans: i32
+    count: dict[i32, i32] = {0: 0}
+    last: i32 = x[0]
+    ct: i32 = 0
+    max_count: i32 = 0
+    c: i32
+    for c in range(k):
+        if x[c] == last:
+            ct += 1
+        else:
+            count[last] = ct
+            last = x[c]
+            ct = 1
+    count[last] = ct
+    for c in range(k):
+        if max_count < count[x[c]]:
+            max_count = count[x[c]]
+            ans = x[c]
+    return ans
+
+# TODO: Uncomment this code when hashing for f64 is implemented
+# @overload
+# def mode(x: list[f64]) -> f64:
+#     """
+#     Returns the mode of a data sequence of numbers
+#     """
+#     k: i32 = len(x)
+#     if k == 0:
+#         raise Exception("No mode for empty data")
+#     ans: f64
+#     count: dict[f64, i32] = {0.0: 0}
```

Comment:
https://stackoverflow.com/a/690791. I think we shouldn't use dicts here and allow for an error tolerance as input. Anyways removing `f64` overloads is better for now, instead of waiting for hash function for `f64` to appear. `49.7999999999` and `49.8` are same or different. See the ambiguity below,

```python
>>> import numpy as np
>>> from scipy import stats
>>> d = np.array([1.7, 2.1, 2.1, 30.0, 49.7999999999, 49.7999999999, 49.7999999999, 49.8, 5.0])
>>> m = stats.mode(d)
>>> m
ModeResult(mode=array([49.8]), count=array([3])) # 49.8 is the mode but then the frequency is wrong.
```

Reply:
I would not use floats as keys, only as values in dicts.

Reply:
We should disallow it in AST->ASR, with a nice error message and explanation.

Reply:
If users really want it, they can cast the f64 to i64 and use that as keys.

Reply:
Yes. If users want to use `f64` keys in LPython they should provide their own hash function and their own equality check to compare floats in case of collisions. Until we support these advanced features an error message at ASR level should be fine.